From a1e4ab8b82429320c5f962038c0f28b874f8ea6f Mon Sep 17 00:00:00 2001
From: Kai Kang <kai.kang@windriver.com>
Date: Thu, 13 Dec 2018 10:55:36 +0800
Subject: [PATCH lttng-tools] Fix: compile fails for x32 arch
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

LTTng-Tools fails to compile for the x32 ABI:

| .../src/common/utils.c: Assembler messages:
| .../src/common/utils.c:1026: Error: register type mismatch for `bsr'
| .../src/common/utils.c:1028: Error: operand type mismatch for `movq'

Add a macro to only use the x86_64 inline assembly version of fls_u64()
when the LP64 ABI is used.

Signed-off-by: Kai Kang <kai.kang@windriver.com>
Signed-off-by: Jérémie Galarneau <jeremie.galarneau@efficios.com>
---
 src/bin/lttng/utils.c | 2 +-
 src/common/utils.c    | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

--- a/src/bin/lttng/utils.c
+++ b/src/bin/lttng/utils.c
@@ -159,7 +159,7 @@
 #define HAS_FLS_U32
 #endif
 
-#if defined(__x86_64)
+#if defined(__x86_64) && defined(__LP64__)
 static inline
 unsigned int fls_u64(uint64_t x)
 {
--- a/src/common/utils.c
+++ b/src/common/utils.c
@@ -1015,7 +1015,7 @@
 #define HAS_FLS_U32
 #endif
 
-#if defined(__x86_64)
+#if defined(__x86_64) && defined(__LP64__)
 static inline
 unsigned int fls_u64(uint64_t x)
 {
